controller/main.go

```diff
@@ -165,14 +166,20 @@ func main() {
 		bgpType = "native"
 	}
 
+	_, present = os.LookupEnv("METALLB_CONTROLLER_METRICS_PORT")
```

Comment:
I'd force the default directly on the parameter instead of refetching the env variable, with a comment about it.

if *controllerMetricsPort = "" {
    *controllerMetricsPort = "0"
}